arangod/Agency/Agent.cpp

```diff
@@ -1136,6 +1139,12 @@ void Agent::run() {
         continue;
       }
 
+      // Challenge leadership:
+      if (challengeLeadership()) {
+        resign();
+        continue;
+      }
+
```

Comment:
Not against this, but would you drop a comment as to why this is an appropriate place to check?

Reply:
I observed that a leader which loses its two followers simply does not resign forever. The Constituent thread does not time out in its usual way when the agent is a leader. This is not good for a split brain scenario, because an isolated leader could continue serving read requests. Note that it cannot serve a write request, since that would involve replicating it to at least one more server. I thought the easiest fix for this would be to put the challengeLeadership and possible resignation *anywhere* in the main loop of the `Agent` thread. This is how it ended up here.

Reply:
Add a quick comment to this fact in the code?